### DISCUSSION TITLE
Using precomputed PCA and  UMAP coordinates

### DISCUSSION THREAD

**Discussion by TarekGharsalli** [theislab/scvelo#722] [General]
Hello, 

Does any one have an idea if we should import PCA coordinates from Seurat object when we want to use  Seurat UMAP coordinates and Seurat clusters in our adata in scvelo ?

Thanks

**Comment by WeilerP** [comment]
@TarekGharsalli, if you simply want to visualize the UMAP embedding, it suffices to store the coordinates calculated in Seurat in your AnnData (`.obsm['X_umap']`) and clusters in `.obs`. If you also want to use the underlying neighbor graph for imputing the data (first order moments), you'd have to store the neighbor graph as well.

**Reply by TarekGharsalli** [reply]
Thanks for your fast reply @Weiler 
I 'm a bit confused because in Seurat it compute the UMAP based on PCA 
So if I run scv.pp.moments which compute pca and neighbor and then I use Seurat UMAP coordinate to visualize it doesn't affect the direction of the arrows because I use a UMAP calculated  on another PCA embedding and not the PCA calculated in scvelo ?
Thanks

**Reply by WeilerP** [reply]
UMAP coordinates are calculated by reducing the dimension using PCA, calculating a neigbor graph on it and then the UMAP embedding. `scv.pp.moments` calculates PCA to compute a neighbor graph based on which you impute the data. If a neighbor graph is already present, it will not (or at least should not) calculate a new one.
If you want to be on the save side, you simply save PCA coordinates, neighbor graph and UMAP coordinates calculated in Seurat in your AnnData.

**Reply by Alvis-Jiang** [reply]
Hi @WeilerP, 

I also had a quenstion, now I have a integrated seurat objective with SCT, and I am able to extract PCA and UMPA from seurat and store them into anndata. For the neighbor graph, I could only export the SNN graph from the seurat and store it as "connectiveness" in the anndata. But I failed to find and export distance from seurat, could please give me some guidance on what neighbor graph should I export from seurat and store them into anndata for scvelo? 

Thanks a lot!

**Comment by TarekGharsalli** [comment]
Thanks a lot @WeilerP  !!!
